rfcs/0000-multiple-control-plane-providers.md

```diff
@@ -0,0 +1,270 @@
+
+```yaml
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant-sample-1
+  namespace: kuadrant-system
+spec:
+  gatewayProvider:
+    group: maistra.io/v2
+    kind: ServiceMeshControlPlane
+    name: basic
+    namespace: istio-system
+---
+apiVersion: kuadrant.io/v1beta1
+kind: Kuadrant
+metadata:
+  name: kuadrant-sample-2
+  namespace: kuadrant-system
+spec:
+  gatewayProvider:
+    group: install.istio.io/v1alpha1
+    kind: IstioOperator
+    name: istiocontrolplane
+    namespace: istio-system
+```
+
+The issue with this simpler approach depends on the definition of "Kuadrant instance". While one instance implies one deployment of Authorino and (more importantly) one deployment of Limitador, sharing a functional component of Kuadrant across services meshes would not be possible. For rate limiting with no support for external storage (nor any kind synchronisation between instances), this limitation translates to no shared counters across different control planes, i.e. each service mesh sees a different count.
+
+Perhaps the described issue is a minor issue or even no issue at all, given the closed aspect of service meshes. I.e., one could argue that a service mesh is supposed to self-contain all the management services it requires, in a way that instances of Authorino and Limitador (and therefore instances of Kuadrant – according to current definition) are not to be shared across different meshes anyway; ergo pointing one and only one gateway provider for each Kuadrant instance would suffice.
+
+If, on the other hand, the premise above is found to be false, then either multiple gateways providers per instance of Kuadrant has to be supported or the definition of "Kuadrant instance" needs to be revised.
+
+<br/>
+
+**Alternative B.** Implicit reference to the gateway provider.
+
+With the proposal coming with [RFC Wiring policies with Kuadrant services](https://github.com/Kuadrant/architecture/pull/5), an alternative way to also link the Kuadrant instances and the corresponding control plane/gateway providers into which the Operator needs to inject configuration could be by following the references between either:
+1. the Kuadrant policies (`RateLimitPolicy` and `AuthPolicy`) and their targeted network resources, that ultimately link all the way with gateways and the gateway providers; or
+2. (more directly) the `Kuadrant` custom resources and their corresponding gateway objects, linked via labels and label selectors (see [architecture#5:124-126](https://github.com/Kuadrant/architecture/blob/9545f760b66ff9aaddeeaab126d5276a086e228c/rfcs/0000-wire-policies-with-backend-services.md?plain=1#L124-L126)).
+
+While option 1 above implies a lazy configuration of the functional components into the gateway providers, option 2 perhaps could be implemented in a rather proactive fashion, such as the one proposed in this RFC.
```

Comment:
not sure I am a fan of the lazy configuration. Or I should say, I like the hands off approach esp for developer exp, but  I think I would want that to be an explicit opt in. It is the role of the  cluster-admin to decide to install kuadrant and it should be up to that persona to decide which gateway providers it applies to. It feels like doing this lazy config moves that reponsibility (perhaps unexpectedly) into the hands of the person defining a RLP for example?